packages/tabs/src/index.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component, Fragment } from "react";
+import { string } from "prop-types";
+
+class Tabs extends Component {
+  constructor(props) {
+    super(props);
+    const tabIds = [],
+      panelIds = [];
+    props.children.forEach(child => {
+      if (child.type.name === "TabBar") {
+        child.props.children.forEach(c => {
+          tabIds.push(c.props.id);
+        });
+      } else if (child.type.name === "TabPanels") {
+        child.props.children.forEach(c => {
+          panelIds.push(c.props.id);
+        });
+      }
+    });
+    this.state = { activeIndex: props.activeIndex, tabIds, panelIds };
+  }
+
+  render() {
+    const { activeIndex, labels, panelIds, tabIds } = this.state;
+    const { activeIndex: startingIndex, ...props } = this.props;
+    return (
+      <div {...props}>
+        {React.Children.map(this.props.children, child => {
+          if (child.type.name === "TabBar") {
+            return React.cloneElement(child, {
+              activeIndex,
+              onClick: activeIndex => {
+                this.setState({ activeIndex });
+              },
+              panelIds
+            });
+          } else if (child.type.name === "TabPanels") {
+            return React.cloneElement(child, { activeIndex, labels, tabIds });
+          }
+        })}
+      </div>
+    );
+  }
+}
+
+Tabs.defaultProps = {
+  activeIndex: 0
+};
+
+let TabBar = ({ activeIndex, onClick, panelIds, ...props }) => {
+  const maxIndex = props.children && props.children.length - 1;
```

Comment:
This can be undefined, won't it be better to use `React.Children.count(children)`?